exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchTraceExporter.java

```diff
@@ -0,0 +1,244 @@
+
+  @GuardedBy("monitor")
+  @Nullable
+  private static Handler handler = null;
+
+  private ElasticsearchTraceExporter() {}
+
+  /**
+   * Creates and registers the ElasticsearchTraceExporter to the OpenCensus.
+   *
+   * @param elasticsearchTraceConfiguration {@link ElasticsearchTraceConfiguration}
+   * @throws MalformedURLException when the Elasticsearch url is invalid.
+   * @throws IllegalStateException if ElasticsearchTraceExporter is already created.
+   * @throws IllegalArgumentException when mandatory parameters in the configuration are missing.
+   * @since 0.20.0
+   */
+  @SuppressWarnings("nullness")
+  public static void createAndRegister(
+      ElasticsearchTraceConfiguration elasticsearchTraceConfiguration)
+      throws MalformedURLException {
+    synchronized (monitor) {
+      Preconditions.checkState(handler == null, "Elasticsearch exporter already registered.");
+      Preconditions.checkArgument(
+          elasticsearchTraceConfiguration != null, "Elasticsearch " + "configuration not set.");
+      Preconditions.checkArgument(
+          elasticsearchTraceConfiguration.getElasticsearchIndex() != null,
+          "Elasticsearch index not specified");
+      Preconditions.checkArgument(
+          elasticsearchTraceConfiguration.getElasticsearchType() != null,
+          "Elasticsearch type not specified");
+      Preconditions.checkArgument(
+          elasticsearchTraceConfiguration.getElasticsearchUrl() != null,
+          "Elasticsearch URL not specified");
+      handler = new ElasticsearchTraceHandler(elasticsearchTraceConfiguration);
+      register(Tracing.getExportComponent().getSpanExporter(), handler);
+    }
+  }
+
+  /**
+   * Registers the ElasticsearchTraceExporter.
+   *
+   * @param spanExporter instance of the {@link SpanExporter} registered.
+   * @param handler instance of the {@link Handler} registered.
+   */
+  public static void register(SpanExporter spanExporter, Handler handler) {
+    spanExporter.registerHandler(REGISTERED_TRACE_EXPORTER_NAME, handler);
+  }
+
+  /**
+   * Unregisters the ElasticsearchTraceExporter from OpenCensus.
+   *
+   * @throws IllegalStateException if ElasticsearchTraceExporter is already unregistered.
+   * @since 0.20.0
+   */
+  public static void unregister() {
+    synchronized (monitor) {
+      Preconditions.checkState(
+          handler != null,
+          "Can't unregister Elasticsearch Trace Exporter which is not registered.");
+      unregister(Tracing.getExportComponent().getSpanExporter());
+      handler = null;
+    }
+  }
+
+  /**
+   * Unregisters the ElasticsearchTraceExporter from OpenCensus.
+   *
+   * @param spanExporter the instance of the {@code SpanExporter} to be unregistered.
+   * @since 0.20.0
+   */
+  public static void unregister(SpanExporter spanExporter) {
+    spanExporter.unregisterHandler(REGISTERED_TRACE_EXPORTER_NAME);
+  }
+
+  static final class ElasticsearchTraceHandler extends SpanExporter.Handler {
```

Comment:
nit: better to split `ElasticsearchTraceHandler` to its own class, similar to other exporters.